src/tls/mitlsffi.h

```diff
@@ -0,0 +1,47 @@
+#ifndef HEADER_MITLS_FFI_H
+#define HEADER_MITLS_FFI_H
+
+typedef struct mitls_state mitls_state;
+
+// Functions exported from libmitls.dll
+//   Functions returning 'int' return 0 for failure, or nonzero for success
+
+// Perform one-time initialization
+extern int  FFI_mitls_init(void);
+
+// Perform one-team termination
+extern void FFI_mitls_cleanup(void);
+
+// Configure miTLS ahead of connecting
+extern int   FFI_mitls_configure(/* out */ mitls_state **state, const char *tls_version, const char *host_name, /* out */ char **outmsg, /* out */ char **errmsg);
+
+// Close a miTLS session - either after configure or connect
+extern void FFI_mitls_close(/* in */ mitls_state *state);
+
+// Callbacks from miTLS to the host application, to send and receive TCP, invoked within FFI_mitls_connect() only
+struct _FFI_mitls_callbacks;
+typedef int (*pfn_FFI_send)(struct _FFI_mitls_callbacks *callbacks, const void *buffer, size_t buffer_size);
```

Comment:
Correct - libcurl allocates a struct on the C heap that contains the _FFI_mitls_callbacks and the additional data it needs in order to send and receive over TCP (such as the socket identifier and CURL session data).  The callback functions use use the callback address to recover all of the data.  Happily, all of the curl side of miTLS is located in one file in the curl repo - lib/vtls/mitls.c, so it's easy to review.